packages/cli/src/commands/create/git.ts

```diff
@@ -6,12 +6,12 @@ import { pkg } from '../../pkg.js';
 import { warnLabel } from '../../utils/messages.js';
 import { runShellCommand } from '../../utils/shell.js';
 import { assertNotCanceled, runTask } from '../../utils/tasks.js';
-import { DEFAULT_VALUES, type CreateOptions } from './options.js';
+import { DEFAULT_VALUES, readFlag, type CreateOptions } from './options.js';
 
 export async function initGitRepo(cwd: string, flags: CreateOptions) {
-  let shouldInitGitRepo = flags.git ?? DEFAULT_VALUES.git;
+  let shouldInitGitRepo = readFlag(flags, 'git');
```

Comment:
I created a small `readFlag()` utility which returns the value of the flag, or if the flag doesn't have a value, it will return the default ONLY if `flags.default` is `true`. It makes the logic a bit easier because you know that if the value is `undefined`, you can prompt the user for the question because it also means `--defaults` was not provided.